### PR TITLE
Check for nil pointer in update rule parameters

### PR DESCRIPTION
**Pull request by o-sama**
Since discussing this with the GitHub folks, I’ve been told that the `update_allows_fetch_and_merge` parameter for the update rule only applies to forked repos. **Note: I was told this is still bugged for forked repos when sending a request through the API, but will be fixed in the near future.**

Since it only applies to forked repos, the API returns a ruleset which contains:
```
{
   "type": “update"
}
```

This doesn't have a parameters field, so when we try to unmarshal the JSON we run into a nil pointer dereference error. No functionality is changed here except to check for a nil pointer before we try to unmarshal the JSON, and if it isn’t included then we just set the parameters to nil.

I didn’t think this needed a change in testing because it doesn’t change any functionality, but I’ll be happy to go and add something if you think otherwise.

@gmlewis I’d really appreciate if this can be part of the upcoming release (which I know is short notice) since I’d love for it to be included in https://github.com/integrations/terraform-provider-github/pull/1808. No worries if it’s too late though, I can just keep it out for now. Thanks!!

### CODE REVIEW

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:122]: Let's please change this to make it easier to read:

```go
if RepositoryRule.Parameters == nil {
  r.Parameters = nil
  return nil
}
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @o-sama!
Please add a new unit test to cover this case.
Yes, we can hold off on the next release until this PR is merged.

File: github/repos_rules.go
Comment (gmlewis) [reply github/repos_rules.go:122]: Right, I understand, but I want to highlight the "happy path" and provide the early return which is idiomatic Go.
Please switch to my suggestion, and feel free to put a blank line after the new block if you think it reads better.

Comment (o-sama) [reply github/repos_rules.go]: Done, also added a bit in the update rule constructor to check for empty params since this specific rule is a bit different than others 🙂

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:124]: One of the points was to remove the "else" and un-indent the following section, please:

```suggestion
		}
```

Comment (o-sama) [reply github/repos_rules.go:124]: You’re totally right, I think I had a brain freeze. Don’t know why I kept an else with a return in there.

**Review by gmlewis — APPROVED** [review]
Thank you, @o-sama !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by o-sama** [comment]
Thanks @gmlewis! sorry about the sloppiness earlier with the requested change 🙏

**Review by liaodaniel — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
> Thanks @gmlewis! sorry about the sloppiness earlier with the requested change 🙏

No problem! Sorry for the delayed responses... Work sometimes gets in the way. 😂

**Comment by gmlewis** [comment]
Thank you, @liaodaniel !
Merging.

**Pull request merged**
